sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -848,4 +851,19 @@ object DDLUtils {
       }
     }
   }
+
+  private[sql] def checkFieldNames(table: CatalogTable): Unit = {
+    val serde = table.storage.serde
+    if (serde == HiveSerDe.sourceToSerDe("orc").get.serde) {
+      OrcFileFormat.checkFieldNames(table.dataSchema)
+    } else if (serde == HiveSerDe.sourceToSerDe("parquet").get.serde) {
+      ParquetSchemaConverter.checkFieldNames(table.dataSchema)
+    } else {
+      table.provider.get.toLowerCase(Locale.ROOT) match {
```

Comment:
`table.provider` could be `None` in the previous versions of Spark. Thus, `.get` is risky.